src/Vegetation/PlantHydraulics.jl

```diff
@@ -116,15 +162,27 @@ for simulation flux of water to/from soil, along roots of different depths,
 along a stem, to a leaf, and ultimately being lost from the system by
 transpiration.
 
-This model can be used in standalone mode by prescribing the transpiration rate
-and soil matric potential at the root tips or flux in the roots, or with a
-dynamic soil model using `ClimaLSM`.
+This model can also be combined with the soil model using ClimaLSM, in which
+case the prognostic soil water content is used to determine root extraction, and
+the transpiration is also computed diagnostically. In  global run with patches
+of bare soil, you can "turn off" the canopy model (to get zero root extraction, zero absorption and
+emission, zero transpiration and sensible heat flux from the canopy), by setting:
+- n_leaf = 1
+- n_stem = 0
+- LAI = SAI = RAI = 0.
+
+A plant model can have leaves but no stem, but not vice versa. If n_stem = 0, SAI must be zero.
```

Comment:
Did you say above that it is actually possible to have stem but no leaves? (I imagine deciduous trees in winter for example).
 
Sorry I may be a bit confused here.

Reply:
oh, good question. In the code right now, we require n_leaf > 0 (and then can set LAI to zero as needed if there is no plant). so we would not support n_stem > 0 but n_leaf = 0. 

I think for deciduous trees we would have n_leaf > 0 but then LAI would vary over time and be very low in the winter. 

Is it possible to have zero LAI but nonzero SAI/RAI? What happens then? are there any fluxes (energy, water, carbon) between the plant and the atmosphere? I think no. and then would there be root extraction still? Right now in this PR we assert that if LAI = 0, SAI = RAI = 0 also.

Reply:
I don't know how it is done in models, but zero LAI with nonzero SAI/RAI definitely happens in winter! 

Photosynthesis is 0, but respiration is not 0 (it is reduced, but not 0). 

Not sure about water - obviously 0 for leaves since no leaves, but maybe reduced in roots (close to 0 but there can be some movement)

About energy, I imagine there is still sensible heat flux, perhaps even latent if trees freeze? (but I imagine that's rare?) 

Not sure what we should do.

Reply:
ok, well, all those fluxes when they get defined will have to be defined so that they have the right behavior. But I guess that is not in the realm of this PR's goals. For now, are these "rules" OK?

1. n_leaf > 0 [you cant have a plant model with a leaf compartment. 
2.  if n_stem > 0, SAI > 0; if n_stem = 0, SAI =0 [ you can have a plant with only roots and leaves, like grass]
3. if LAI > 0, RAI must be > 0
No check required for the following:
4. you can have LAI zero but SAI and RAI nonzero - e.g. deciduous trees in winter. right now these would have zero transpiration but possibly some water flux between roots/stem and soil.
5. If we want to "turn off" the canopy, we set RAI = SAI = LAI = 0, but at a minimum, the leaf compartment is always there.
Is that ok @juliasloan25 @AlexisRenchon?

Reply:
[note - I realized that this LAI consistency check function will have to be refactored once we have LAI read in from a file or varying in time, because we wont be passing the values in as fixed float parameters this way....so im sort of on the fence about if we even need to include this function? But perhaps it is good to include it for now to codify these rules and then we can adapt as needed when we refactor]

Reply:
that's ok with me =)